zproject/backends.py

```diff
@@ -1186,19 +1186,15 @@ def social_associate_user_helper(backend: BaseAuth, return_data: Dict[str, Any],
     full_name = kwargs['details'].get('fullname')
     first_name = kwargs['details'].get('first_name', '')
     last_name = kwargs['details'].get('last_name', '')
-    if full_name is None:
-        if not first_name and not last_name:
-            # We need custom code here for any social auth backends
-            # that don't provide name details feature.
-            if (backend.name == 'apple'):
-                # Apple authentication provides the user's name only
-                # the very first time a user tries to login.  So if
-                # the user aborts login or otherwise is doing this the
-                # second time, we won't have any name data.  We handle
-                # this by setting full_name to be the empty string.
-                full_name = ""
-            else:
-                raise AssertionError("Social auth backend doesn't provide name")
+    if all(name is None for name in [full_name, first_name, last_name]) and backend.name != "apple":
```

Comment:
this won't work correctly if e.g. first_name is the empty string rather than None. We want `not name` I think.

Reply:
Ohh, I used `is None` thinking we'll restrict `full_name = ''` case with `not name`.
Can we change
```
    first_name = kwargs['details'].get('first_name', '')
    last_name = kwargs['details'].get('last_name', '')
```
to 
```
    first_name = kwargs['details'].get('first_name', '')
    last_name = kwargs['details'].get('last_name', '')
```
so that this nice `if` line will still be correct?